### PR TITLE
[AMS-Refactor] fix runtime bug and add unit test

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
refactor ams

## Brief change log

1.fix runtime bug
2.add mix-format table optimize unit test

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by hameizi** [comment]
resolve in https://github.com/NetEase/arctic/pull/1459

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 1]
```diff
@@ -159,14 +160,19 @@ private OptimizingQueue getQueueByToken(String token) {
 
   @Override
   public List<OptimizerInstance> listOptimizers() {
-    return optimizingQueueByGroup.values().stream()
+    return optimizingQueueByGroup.values()
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:163]: I don't think the service layer needs to guarantee the order of results. The order of results should be flexibly set by the Controller layer itself.

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 2]
```diff
@@ -159,14 +160,19 @@ private OptimizingQueue getQueueByToken(String token) {
...
+        .stream()
         .flatMap(queue -> queue.getOptimizers().stream())
+        .sorted(Comparator.comparingLong(OptimizerInstance::getStartTime).reversed())
         .collect(Collectors.toList());
   }
 
   @Override
   public List<OptimizerInstance> listOptimizers(String group) {
-    return getQueueByGroup(group).getOptimizers();
+    return getQueueByGroup(group).getOptimizers()
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:172]: I don't think the service layer needs to guarantee the order of results. The order of results should be flexibly set by the Controller layer itself.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java [hunk 3]
```diff
@@ -92,9 +123,8 @@ public void getTableDetail(Context ctx) {
     String tableMame = ctx.pathParam("table");
 
     Preconditions.checkArgument(
-        catalog != null && database != null && tableMame != null,
-        "catalog.dabatase.tableName can not be null in any element");
-    // getRuntime table from catalog
+        !catalog.isEmpty() && !database.isEmpty() && !tableMame.isEmpty(),
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java:126]: Use org.apache.commons.lang3.StringUtils#isNotBlank

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/SettingController.java [hunk 4]
```diff
@@ -58,19 +58,6 @@ public SettingController(Configurations serviceConfig, OptimizerManager optimize
...
-    putSetting(config, ArcticManagementConf.HTTP_SERVER_PORT.key(),
-        serviceConfig.get(ArcticManagementConf.HTTP_SERVER_PORT));
-    putSetting(config, ArcticManagementConf.DB_TYPE.key(), serviceConfig.get(ArcticManagementConf.DB_TYPE));
-    putSetting(config, ArcticManagementConf.DB_CONNECTION_URL.key(),
-        serviceConfig.get(ArcticManagementConf.DB_CONNECTION_URL));
-    putSetting(config, ArcticManagementConf.DB_CONNECTION_URL.key(),
-        serviceConfig.get(ArcticManagementConf.DB_CONNECTION_URL));
-
     serviceConfig.toMap().forEach((k, v) -> {
       if (!config.containsKey(k)) {
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/SettingController.java:62]: It doesn't seem necessary to put the configuration into the `config` variable again here. Instead, we can directly sort the configuration and put it into the "result" variable below.

File: ams/server/src/main/java/com/netease/arctic/server/table/executor/SnapshotsExpiringExecutor.java [hunk 5]
```diff
@@ -96,7 +96,7 @@ public void execute(TableRuntime tableRuntime) {
         return;
       }
 
-      expireArcticTable(arcticTable, tableRuntime);
+      // expireArcticTable(arcticTable, tableRuntime);
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/table/executor/SnapshotsExpiringExecutor.java:99]: Remove comment

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java [hunk 6]
```diff
@@ -158,69 +188,85 @@ public void getTableDetail(Context ctx) {
...
+      List<AMSColumnInfo> partitionColumnInfos = transformHiveSchemaToAMSColumnInfo(hiveTable.getPartitionKeys());
+      hiveTableInfo = new HiveTableInfo(tableIdentifier, TableMeta.TableType.HIVE, schema, partitionColumnInfos,
+          new HashMap<>(), hiveTable.getCreateTime());
+    } catch (Exception e) {
+      LOG.error("Failed to getRuntime hive table info", e);
+      ctx.json(new ErrorResponse(HttpCode.BAD_REQUEST, "Failed to getRuntime hive table info", ""));
+      return;
+    }
+    ctx.json(OkResponse.of(hiveTableInfo));
     ctx.json(OkResponse.ok());
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java:218]: Is this line of code redundant?

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java [hunk 7]
```diff
@@ -158,69 +188,85 @@ public void getTableDetail(Context ctx) {
...
+    String table = ctx.pathParam("table");
+    Preconditions.checkArgument(
+        !catalog.isEmpty() && !db.isEmpty() && !table.isEmpty(),
+        "catalog.database.tableName can not be empty in any element");
+    UpgradeHiveMeta upgradeHiveMeta = ctx.bodyAsClass(UpgradeHiveMeta.class);
+
+    ArcticHiveCatalog arcticHiveCatalog
+        = (ArcticHiveCatalog) CatalogLoader.load(String.join("/", AmsUtil.getAMSThriftAddress(serviceConfig), catalog));
+    try {
+      new Thread(() -> {
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java:236]: Will changing the upgrade process to asynchronous execution affect the user experience? For example, users may not be able to perceive the upgrade results correctly.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java [hunk 8]
```diff
@@ -158,69 +188,85 @@ public void getTableDetail(Context ctx) {
...
+                  .stream()
+                  .map(UpgradeHiveMeta.PrimaryKeyField::getFieldName)
+                  .collect(Collectors.toList()), upgradeHiveMeta.getProperties());
+          upgradeRunningInfo.get(tableIdentifier).setStatus(UpgradeStatus.SUCCESS.toString());
+        } catch (Throwable t) {
+          LOG.error("Failed to upgrade hive table to arctic ", t);
+          upgradeRunningInfo.get(tableIdentifier).setErrorMessage(AmsUtil.getStackTrace(t));
+          upgradeRunningInfo.get(tableIdentifier).setStatus(UpgradeStatus.FAILED.toString());
+        } finally {
+          executor.schedule(
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java:251]: Why is it necessary to delay deleting the upgrade information here?

File: ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/OptimizingMapper.java [hunk 9]
```diff
@@ -110,20 +114,22 @@ List<TableOptimizingProcess> selectSuccessOptimizingProcesses(
           "end_time, status, fail_reason, optimizer_token, thread_id, rewrite_output, metrics_summary) VALUES ",
       "<foreach collection='taskRuntimes' item='taskRuntime' index='index' separator=','>",
       "(#{taskRuntime.taskId.processId}, #{taskRuntime.taskId.taskId}, #{taskRuntime.retry}," +
-          " #{taskRuntime.tableId}, #{taskRuntime.partition}, #{taskRuntime.startTime}," +
-          " #{taskRuntime.endTime}, #{taskRuntime.status}, #{taskRuntime.failReason}," +
-          " #{taskRuntime.optimizingThread.token}, #{taskRuntime.optimizingThread.threadId}," +
-          " #{taskRuntime.outputBytes," +
-          " typeHandler=com.netease.arctic.server.persistence.converter.JsonSummaryConverter}," +
+          " #{taskRuntime.tableId}, #{taskRuntime.partition}, #{taskRuntime.startTime, typeHandler=com.netease.arctic" +
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/OptimizingMapper.java:117]: I suggest adjusting the line breaks to ensure that the content of the typeHandler appears completely on one line.

File: ams/server/src/main/java/com/netease/arctic/server/table/executor/SnapshotsExpiringExecutor.java (hunk 5 above)
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/table/executor/SnapshotsExpiringExecutor.java:99]: Why comment the code here?

File: core/src/test/java/com/netease/arctic/IcebergTableTestHelper.java [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.ArrayUtil;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class IcebergTableTestHelper {
```
Comment (zhoujinsong) [inline core/src/test/java/com/netease/arctic/IcebergTableTestHelper.java:34]: This helper seems containe duplicate codes with `com.netease.arctic.io.DataTestHelpers`

**Review by shidayang — CHANGES_REQUESTED** [review]

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/SettingController.java (hunk 4 above)
Comment (hameizi) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/SettingController.java:62]: will fix

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java (hunk 2 above)
Comment (hameizi) [reply ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:172]: ok, i will resolve
